src/entries/background/procedures/popup/index.ts

```diff
@@ -0,0 +1,22 @@
+import { RPCHandler } from '@orpc/server/message-port';
+
+import { popupOs } from './os';
+import { walletRouter } from './wallet';
+
+export const popupRouter = popupOs.router({
+  wallet: walletRouter,
+});
+
+export type PopupRouter = typeof popupRouter;
+
+export function startPopupRouter() {
+  const handler = new RPCHandler(popupRouter);
+
+  chrome.runtime.onConnect.addListener((port) => {
+    handler.upgrade(port, {
+      context: {
+        sender: port.sender,
+      },
+    });
+  });
+}
```

Comment:
I do think it would be helpful if we consolidate some of this messager boilerplate into a tight core module. The entrypoint naming scheme falls apart when we have a popupRouter inside the background script, and a background handler in the popup that back-references the popup exports.

Reply:
The `popupOs` naming scheme is also a bit confusing